src/NBitcoin/ChainIndexer.cs

```diff
@@ -66,9 +66,9 @@ public void Initialize(ChainedHeader chainedHeader)
                     this.blocksById.Add(iterator.HashBlock, iterator);
                     this.blocksByHeight.Add(iterator.Height, iterator);
 
-                    if (chainedHeader.Height == 0)
+                    if (iterator.Height == 0)
                     {
-                        if (chainedHeader.HashBlock != iterator.HashBlock)
+                        if (this.Network.GenesisHash != iterator.HashBlock)
```

Comment:
The `ChainIndexer` is not always used with a first block that matches the genesis block:

```
        private void CanVerifySequenceLockCore(Sequence[] sequences, int[] prevHeights, int currentHeight, DateTimeOffset first, bool expected, SequenceLock expectedLock)
        {
            Network network = this.stratisMain;
            BlockHeader blockHeader = network.Consensus.ConsensusFactory.CreateBlockHeader();
            blockHeader.BlockTime = first;

            var chain = new ChainIndexer(network, new ChainedHeader(blockHeader, blockHeader.GetHash(), 0));
```

That means these tests will have to change or revert this.

Reply:
Yeah we probably need to change the test, I will have a look.